### PR TITLE
fix container removal if auto_remove was enabled

### PR DESCRIPTION
**Pull request by bmiguel-teixeira**
### What does this PR do?
Corrects a bug in the container stop & removal flow. It the container was started with the AutoRemove flag  (--rm), when stopped, the docker engine will automatically remove it, and salt's rm function will throw a 404 when removing.

### What issues does this PR fix or reference?
Did not open an issue.

### Previous Behavior
`salt myminion docker.rm mycontainer stop=True`
This would throw 404 if `mycontainer` had the AutoRemove flag enabled as stopping the container would imply auto removal.

### New Behavior
`salt myminion docker.rm mycontainer stop=True`
If `stop=True`, we check the container's AutoRemove flag and do not remove the container if the flag is set to True. The docker engine will take care of that.

No longer throws 404 if `mycontainer` had the AutoRemove flag enabled.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/modules/dockermod.py [hunk 1]
```diff
@@ -3807,9 +3808,12 @@ def rm_(name, force=False, volumes=False, **kwargs):
             'remove this container'.format(name)
         )
     if stop_ and not force:
+        auto_remove = inspect_container(name)['HostConfig']['AutoRemove']
```
Comment (terminalmage) [inline salt/modules/dockermod.py]: To future-proof us against Docker changing inspect return data structure, we should catch on `KeyError` here:

```python
        inspect_results = inspect_container(name)
        try:
            auto_remove = inspect_results['HostConfig']['AutoRemove']
        except KeyError:
            log.error(
                'Failed to find AutoRemove in inspect results, Docker API may '
                'have changed. Full results: %s', inspect_results
            )
            # Assume False and attempt to remove the container below
            auto_remove = False
```

**Review by terminalmage — CHANGES_REQUESTED** [review]
@rallytime once this is merged we'll need a backport to 2017.7.

File: salt/modules/dockermod.py (hunk 1 above)
Comment (bmiguel-teixeira) [reply salt/modules/dockermod.py]: agreed with eafp

Comment (terminalmage) [reply salt/modules/dockermod.py]: Good point, I forgot you had set it above. :+1:

**Review by terminalmage — APPROVED** [review]

**Pull request merged**